src/ipcEventsAddon.py

```diff
@@ -38,18 +43,95 @@ def done(self):
         self._sendIpcMessage({"status": "done"})
 
     def client_connected(self, client):
-        self._sendIpcMessage({"status": "clientConnected", "context": {"address": client.peername} })
+        self._sendIpcMessage({"status": "clientConnected", "context": {"address": human.format_address(client.peername), "client": clientToDict(client)}})
 
     def client_disconnected(self, client):
-        self._sendIpcMessage({"status": "clientDisconnected", "context": {"address": client.peername} })
+        self._sendIpcMessage({"status": "clientDisconnected", "context": {"address": human.format_address(client.peername), "client": clientToDict(client)}})
 
     def tls_failed_client(self, data):
-        self._sendIpcMessage({"status": "tlsFailed", "context": {"clientAddress": data.context.client.peername, "serverAddress": data.context.server.address, "error": data.conn.error }})
+        self._sendIpcMessage({"status": "tlsFailed", "context": {"error": data.conn.error, **tlsDataToDict(data)}})
 
     def tls_established_client(self, data):
-        self._sendIpcMessage({"status": "tlsEstablished", "context": {"clientAddress": data.context.client.peername, "serverAddress": data.context.server.address}})
+        self._sendIpcMessage({"status": "tlsEstablished", "context": tlsDataToDict(data)})
 
     def server_changed(self):
         self._sendIpcMessage({"status": "proxyChanged", "servers":  [s.to_json() for s in self.proxyserver.servers]})
```

Comment:
This is kinda inconsistent now, since you convert the other instances to a `dict` using custom methods. I think you should do this here, too.

Reply:
Also, for some reason, the data isn't wrapped in a `context` here, either. Guess I'll change that, too then.